### DISCUSSION TITLE
Object queries

### DISCUSSION THREAD

**Discussion by FrozenEmpire** [ChilliHugger/The-Lords-Of-Midnight#183] [Code Algorithm Questions] [answered]
Hi Chris,

Can you have a look into tblObjects.tsv and let me know whether the following entries are correct
118 | OB_SWORD_THORTHAK                     > should be a SPEAR ?
124 | OB_HAMMER_TORORTHANE              > should be a BOW ?

Thanks

**Comment by IcemarkUK** [comment, marked as answer]
Ok, working this through...

Table 0xa280 (0x0d) is used for both characters and objects. If you check the [data format](https://www.icemark.com/dataformats/revenge/index.html) you will see that it is split in to two nibbles and the nibbles are either

Race (lo) and Orders (hi) for characters or
Use (lo) and Type (hi) for objects.

** The dataformat description is WRONG here. Only the lo nibble is used as both Use and Type for objects!!

| Index | value |
| - | - |
| 0x00 | persuasion |
| 0x01 | persuasion |
| 0x02 | persuasion |
| 0x03 | persuasion |
| 0x04 | protection |
| 0x05 | protection |
| 0x06 | battle |
| 0x07 | battle |
| 0x08 | battle |
| 0x09 | battle |
| 0x0a | battle |
| 0x0b | battle |
| 0x0c | swiftness |
| 0x0d | battle |
| 0x0e | presuasion |
| 0x0f | battle |

| Index | value |
| - | - |
| 0x00 | crown |
| 0x01 | crown |
| 0x02 | crown |
| 0x03 | crown |
| 0x04 | runes |
| 0x05 | runes |
| 0x06 | bow |
| 0x07 | bow |
| 0x08 | spear |
| 0x09 | spear |
| 0x0a | sword |
| 0x0b | sword |
| 0x0c | spell |
| 0x0d | hammer |
| 0x0e | crown |
| 0x0f | axe |

These are the 128 objects.

```
000 0000680 30 32 3c 34 3e 3d 3d 3a 3b 3a 3a 3a 3a 3d 3d 3a 
016 0000690 3d 3b 3a 3a 3a 3d 3d 3a 3a 3a 3a 3a 1b 3d 3a 3a 
032 00006a0 1a 3a 3a 3a 3d 3d 3b 3a 3a 3d 3a 3a 3a 3a 3a 3a 
048 00006b0 3a 3d 3f 3a 3a 3a 3d 3d 3a 3a 3d 3d 3a 3b 1f 3d 
064 00006c0 3f 3a 3a 3a 1d 3a 3f 3d 3f 3d 3d 3f 38 38 3f 3f 
080 00006d0 3f 3f 3f 3f 38 3f 3f 3f 3f 38 3f 3f 38 38 3f 3f 
096 00006e0 39 36 38 36 36 38 37 36 36 38 16 37 18 36 36 37 
112 00006f0 36 37 39 38 36 36 37 38 36 38 36 3d 3b 3b 38 36 
```

The spreadsheet is offset by 22 from the original: so 118 = 96, and 124 =  102

So 118 = 0x39 therefore 0x09  Spear / Battle
And 124 = 0x37 therefore 0x07 Bow / Battle

Therefore you are correct. The description of these objects is incorrect.

**Reply by IcemarkUK** [reply]
I have double checked the objects and these are the only two that are incorrect.

**Reply by FrozenEmpire** [reply]
Thanks Chris,

I suspected so on the assumption that the names were correct therefore the weapon types were not characteristic of the race
